src/main/java/com/minecolonies/coremod/client/render/TileEntityInfoPosterRenderer.java

```diff
@@ -0,0 +1,212 @@
+import net.minecraftforge.client.model.pipeline.LightUtil;
+import net.minecraftforge.common.model.IModelState;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.lwjgl.opengl.GL11;
+
+import java.util.List;
+
+@SideOnly(Side.CLIENT)
+public class TileEntityInfoPosterRenderer extends TileEntitySpecialRenderer<TileEntityInfoPoster>
+{
+    private static final ResourceLocation SIGN_TEXTURE = new ResourceLocation(Constants.MOD_ID, "textures/blocks/info/info_poster.png");
+
+    /**
+     * The ModelSign instance for use in this renderer
+     */
+    private IBakedModel model = null;
+
+    @Override
+    public void renderTileEntityAt(final TileEntityInfoPoster te, final double x, final double y, final double z, final float partialTicks, final int destroyStage)
+    {
+        if(model == null)
+        {
+            model = loadModel();
+        }
+
+        final World world = te.getWorld();
+        final IBlockState state = world.getBlockState(te.getPos());
+        final BlockPos pos = te.getPos();
+        final IBlockState actualState = state.getBlock().getActualState(state, world, pos);
+
+        int k = te.getBlockMetadata();
+        float f2 = 0.0F;
+
+        if (k == 2)
+        {
+            f2 = 180.0F;
+        }
+
+        if (k == 4)
+        {
+            f2 = 90.0F;
+        }
+
+        if (k == 5)
+        {
+            f2 = -90.0F;
+        }
+        GlStateManager.rotate(-f2, 0.0F, 1.0F, 0.0F);
+
+        for (final BlockRenderLayer layer : BlockRenderLayer.values())
+        {
+            if (ModBlocks.blockInfoPoster.canRenderInLayer(actualState, layer))
+            {
+                final EntityPlayer player = Minecraft.getMinecraft().thePlayer;
+                final double dx = player.lastTickPosX + (player.posX - player.lastTickPosX) * partialTicks;
+                final double dy = player.lastTickPosY + (player.posY - player.lastTickPosY) * partialTicks;
+                final double dz = player.lastTickPosZ + (player.posZ - player.lastTickPosZ) * partialTicks;
+
+                GlStateManager.pushMatrix();
+                GlStateManager.translate(pos.getX() - dx, pos.getY() - dy, pos.getZ() - dz);
+
+                RenderHelper.disableStandardItemLighting();
+
+                if (layer == BlockRenderLayer.CUTOUT)
+                {
+                    Minecraft.getMinecraft().getTextureManager().getTexture(SIGN_TEXTURE).setBlurMipmap(false, false);
+                }
+
+                GlStateManager.color(1F, 1F, 1F, 1F);
+
+                final int alpha = ((int) (1.0D * 0xFF)) << 24;
+
+                GlStateManager.enableBlend();
+                GlStateManager.enableTexture2D();
+
+                GlStateManager.blendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
+                GlStateManager.colorMask(false, false, false, false);
+                this.renderModel(world, model, pos, alpha);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "renderModel()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)